### PR TITLE
Add way to pass chip description file

### PR DESCRIPTION
**Pull request by KOBA789**
Fixes: https://github.com/knurling-rs/probe-run/issues/291

I need the way to pass my own chip description file to probe-run because my target chip is a original soft processor core implemented on a FPGA.

There are other ideas on #291 but I wrote the simplest implementation for now.

### CODE REVIEW

File: src/cli.rs [hunk 1]
```diff
@@ -27,6 +27,9 @@ pub(crate) struct Opts {
     #[structopt(long, required_unless_one(&["list-chips", "list-probes", "version"]), env = "PROBE_RUN_CHIP")]
     chip: Option<String>,
 
+    #[structopt(name = "chip description file path", long = "chip-description-path")]
+    pub(crate) chip_description_path: Option<PathBuf>,
```
Comment (Urhengulas) [inline src/cli.rs:31]:
```suggestion
    /// Path to chip description file, in YAML format.
    #[structopt(long)]
    pub(crate) chip_description_path: Option<PathBuf>,
```

**Review by Urhengulas — CHANGES_REQUESTED** [review]
Thank you for the PR @KOBA789!

When specifying the `--chip-description-path`, does a user still need to specify the `--chip` additionally? I don't ask what `clap` requires, but `probe-rs`.

File: src/main.rs [hunk 2]
```diff
@@ -52,6 +52,9 @@ fn run_target_program(elf_path: &Path, chip_name: &str, opts: &cli::Opts) -> any
     let elf_bytes = fs::read(elf_path)?;
     let elf = &Elf::parse(&elf_bytes)?;
 
+    if let Some(ref cdp) = opts.chip_description_path {
+        probe_rs::config::add_target_from_yaml(Path::new(cdp))?;
+    }
```
Comment (Urhengulas) [inline src/main.rs:57]:
```suggestion
    if let Some(cdp) = &opts.chip_description_path {
        probe_rs::config::add_target_from_yaml(Path::new(cdp))?;
    }
```

**Comment by KOBA789** [comment]
Thank you for your review.

> When specifying the --chip-description-path, does a user still need to specify the --chip additionally? I don't ask what clap requires, but probe-rs.

A chip description file may contains some variants. I think `--chip` is necsessary to specify which the variant to use.

**Comment by Urhengulas** [comment]
> A chip description file may contains some variants. I think `--chip` is necsessary to specify which the variant to use.

Makes sense, I just wanted to check with you.

bors r+

**Pull request merged**